pandas-stubs/core/frame.pyi

```diff
@@ -104,7 +106,22 @@ class DataFrame:
     def count(self) -> Series: ...
     def drop(self, index: Union[List[_str], Index], axis: _AxisType = ...) -> DataFrame: ...
     def drop_duplicates(self, keep: Union[_str, bool] = ...) -> DataFrame: ...
-    def dropna(self, axis: int = ..., how: _str = ...) -> DataFrame: ...
+    @overload
+    def dropna(
+        self,
+        axis: int = ...,
+        how: _str = ...,
+        subset: _ColSubsetType = ...,
+        inplace: Literal[False] = ...,
+    ) -> DataFrame: ...
+    @overload
+    def dropna(
+        self,
+        axis: int = ...,
+        how: _str = ...,
+        subset: _ColSubsetType = ...,
+        inplace: Literal[True] = ...,
```

Comment:
```suggestion
        inplace: Literal[True],
```


Hi, thanks for the PR :)
With the current type signatures mypy has no way of distinguishing between the 2 `dropna` signatures as both have default values on the only argument which changes (`inplace`).
We can disambiguate by making `inplace` only be required when it's not the default value of `False`.

Not sure that makes sense, let me know if it just sounds like a ramble 😅 

Also, if you accept this suggestion, you'll need to re-order the args  so that inplace comes after self.